go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/cilium/proxy
 
-go 1.21
+go 1.22
```

Comment:
why not pinning it to a specific patch version?

Reply:
A few weeks back I was having issue with go version mismatch with cilium/cilium in CI, this is probably just a small fix, but I didn't look further. 

Having a generic 1.22 will work well with any go pinned version from cilium/cilium.